app/src/features/surveys/view/SurveyAnimals.tsx

```diff
@@ -99,6 +107,22 @@ const SurveyAnimals: React.FC = () => {
     ]
   };
 
+  const obtainAnimalFormInitialvalues = (mode: ANIMAL_FORM_MODE): IAnimal => {
+    switch (mode) {
+      case ANIMAL_FORM_MODE.ADD:
+        return AnimalFormValues;
+      case ANIMAL_FORM_MODE.EDIT: {
+        const existingCritter = critterData?.find(
+          (critter: IDetailedCritterWithInternalId) => currentCritterbaseCritterId === critter.critter_id
+        );
+        if (!existingCritter) {
+          throw Error('This should not be reachable.');
```

Comment:
I don't think I'd find this error very helpful if I saw this in prod. Can we just `break` instead? Or is there an otherwise more graceful way to recover?

Reply:
It also crashes the app which is extra unhelpful. I think throwing here is appropriate but I implemented a way to handle it, let me know what you think.